ngapp/src/app/student-components/dashboard/dashboard.component.ts

```diff
@@ -440,9 +466,14 @@ export class DashboardComponent implements OnInit {
   }
 
   // save story and set next modal choice to true 
-  saveModal() {
-    this.saveStory();
-    this.modalChoice.next(true);
+  async saveModal() {
+    try {
+      await this.saveStory();
+      this.modalChoice.next(true);
+    } catch (error) {
+      window.alert('You\'re story was not saved. You should copy your story to another program to save it. Otherwise it may be lost.'); 
```

Comment:
Typo nit: (although this'll probably get localised at some point anyway!)

```suggestion
      window.alert('Your story was not saved. You should copy your story to another program to save it. Otherwise it may be lost.'); 
```